react/README.md

```diff
@@ -18,35 +18,25 @@
 
 ## Basic Rules
 
-  - Only include one React component per file.
-  - Always use JSX syntax.
-  - Do not use `React.createElement` unless you're initializing the app from a file that is not JSX.
+  - [1.1](#1.1) <a name='1.1'></a> **File configuration**
 
-## Class vs React.createClass
+    + Only include one React component per file, with the exception of components generated by `styled-components`.
+    + Components generated with `styled-components` are located in the same file as the component they are used in.
+  
+  - [1.2](#1.2) <a name='1.2'></a> **Code conventions**
 
-  - Use class extends Component always.
-
-  ```javascript
-  // bad
-  const Listing = createClass({
-    render() {
-      return <div />;
-    }
-  });
-
-  // good
-  class Listing extends React.Component {
-    render() {
-      return <div />;
-    }
-  }
-  ```
+    + Always use JSX syntax.
+    + Prefer the use of functional components when state and lifecycle methods are not needed.
```

Comment:
We should make mention that functional components don't support refs unless you are using React 16.3.0 or later which has the forwardRef API (https://reactjs.org/blog/2018/03/29/react-v-16-3.html#forwardref-api).

Reply:
yup, ill add some verbiage in there